dali/kernels/transpose/transpose_gpu_impl.cuh

```diff
@@ -25,6 +25,20 @@ namespace dali {
 namespace kernels {
 namespace transpose_impl {
 
+/*
+The transpositions are implemented by calculating the input and output strides and either
```

Comment:
I would add more explanations of the ideas behind the applied algorithms, you can refer to https://arxiv.org/ftp/arxiv/papers/1705/1705.01598.pdf and write a couple of sentences.